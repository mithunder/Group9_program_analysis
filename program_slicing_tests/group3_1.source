module group31 :
    read n;
    total:=3;
    x:=1;
    do
        x<=total-2 ->
        {
            y:=1;
            do
                y<=total-x-1 ->
                    {
                    	z:=total-x-y;
	                    xpn:=1;
	                    i:=1;
	                    do
	                        i<n -> {xpn:=xpn*i; i:=i+1}
	                    od;
@programslicing="true"
	                    ypn:=1;
	                    i:=1;
	                    do
	                        i<n -> {ypn:=ypn*i; i:=i+1}
	                    od;
	                    zpn:=1;
	                    i:=1;
	                    do
	                        i<n -> {zpn:=zpn*i; i:=i+1}
	                    od;
	                    if
	                        exp+ypn=zpn -> write n
	                    fi;
	                    y:=y+1
	                }
            od;
        	x:=x+1
        }
    od
end
