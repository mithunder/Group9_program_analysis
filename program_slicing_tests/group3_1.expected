module group31 :
total := 3;
x := 1;
do
   (x<=(total-2)) ->
      y := 1;
      do
         (y<=((total-x)-1)) ->
            ypn := 1;
            y := (y+1)
      od;
      x := (x+1)
od
end
